### PR TITLE
feature/remove-age

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- #25 

**This PR will result in the following new package version:**

- v0.7.0 since customers may need to resync

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## Breaking Changes
> - In the [July 2023 update](https://fivetran.com/docs/connectors/applications/tiktok-ads/changelog#july2023) for the `ADGROUP_HISTORY` table, the `age` column was renamed to `age_groups`.
>   - Previously, we coalesced these two columns in the `stg_tiktok_ads__ad_group_history` model to account for connectors using the old naming convention. However, due to inconsistent data types, we can no longer use this approach.
>   - As a result, the coalesced field has been removed in favor of the `age_groups` column.
>   - If necessary, you can populate historical data in the `age_groups` column by performing a resync of the `ADGROUP_HISTORY` table, since TikTok provides all data regardless of the previous sync state.
>   - For more details, see the [DECISIONLOG entry](https://github.com/fivetran/dbt_tiktok_ads_source/blob/main/DECISIONLOG.md).

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [ ] ~Detailed validation steps have been provided below~

### Detailed Validation
Please share any and all of your validation steps:

- Minor update: coalesce was removed

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+# dbt_tiktok_ads_source v0.7.0
+[PR #28](https://github.com/fivetran/dbt_tiktok_ads_source/pull/28) includes the following updates:
+
+## Breaking Changes
+- In the [July 2023 update](https://fivetran.com/docs/connectors/applications/tiktok-ads/changelog#july2023) for the `ADGROUP_HISTORY` table, the `age` column was renamed to `age_groups`.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:5]: Just a small suggestion to clarify the connector update.
```suggestion
- In the [July 2023 TikTok Ads connector update](https://fivetran.com/docs/connectors/applications/tiktok-ads/changelog#july2023) for the `ADGROUP_HISTORY` table, the `age` column was renamed to `age_groups`.
```

File: DECISIONLOG.md [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
 
-This is a reason why we have broken out the ad reporting packages into separate hierarchical end models (Ad, Ad Group, Campaign, and more). Because if we only used ad-level reports, we could be missing data.s
\ No newline at end of file
+This is a reason why we have broken out the ad reporting packages into separate hierarchical end models (Ad, Ad Group, Campaign, and more). Because if we only used ad-level reports, we could be missing data.
+
+## `age_groups` and `age` columns
+In the [July 2023 update](https://fivetran.com/docs/connectors/applications/tiktok-ads/changelog#july2023), we renamed the `age` column in the `ADGROUP_HISTORY` table to `age_groups`. To maintain compatibility with connectors using the old column name, our data models previously coalesced the `age` and `age_groups` columns. However, due to inconsistent data types between `age` and `age_groups`, this approach began causing errors.
```
Comment (fivetran-joemarkiewicz) [inline DECISIONLOG.md:9]: Same small suggestion from CHANGELOG
```suggestion
In the [July 2023 TikTok Ads connector update](https://fivetran.com/docs/connectors/applications/tiktok-ads/changelog#july2023), we renamed the `age` column in the `ADGROUP_HISTORY` table to `age_groups`. To maintain compatibility with connectors using the old column name, our data models previously coalesced the `age` and `age_groups` columns. However, due to inconsistent data types between `age` and `age_groups`, this approach began causing errors.
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,14 @@
+# dbt_tiktok_ads_source v0.7.0
+[PR #28](https://github.com/fivetran/dbt_tiktok_ads_source/pull/28) includes the following updates:
+
+## Breaking Changes
+- In the [July 2023 TikTok Ads connector update](https://fivetran.com/docs/connectors/applications/tiktok-ads/changelog#july2023) for the `ADGROUP_HISTORY` table, the `age` column was renamed to `age_groups`.
+  - Previously, we coalesced these two columns in the `stg_tiktok_ads__ad_group_history` model to account for connectors using the old naming convention. However, due to inconsistent data types, we can no longer use this approach.
```
Comment (fivetran-jamie) [inline CHANGELOG.md:6]: Should we include the specific data types (strings and jsons right)?

File: DECISIONLOG.md [hunk 4]
```diff
@@ -1,4 +1,21 @@
...
+- The `age` column has been removed from the `stg_tiktok_ads__ad_group_history` model and the `fill` staging macro for the ad group history table.
+- Models now rely exclusively on the `age_groups` column.
+
+### Impact
+- Customers using the `age` column will no longer have access to this field in the models.
+- The change simplifies logic and resolves data inconsistency issues caused by merging fields with different data types.
+
+### Action for Customers Requiring Historical `age` Data
+Customers who still need the historical `age` column data can:
+1. Resync the `ADGROUP_HISTORY` table in their TikTok Ads connector. TikTok provides all historical data in the `age_groups` column, allowing the data to be fully populated.
```
Comment (fivetran-jamie) [inline DECISIONLOG.md:21]: Is there supposed to be a second option? If not, we should prob remove the `1.`

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,14 @@
...
+  - As a result, the coalesced field has been removed in favor of the `age_groups` column.
+  - If necessary, you can populate historical data in the `age_groups` column by performing a resync of the `ADGROUP_HISTORY` table, since TikTok provides all data regardless of the previous sync state.
+  - For more details, see the [DECISIONLOG entry](https://github.com/fivetran/dbt_tiktok_ads_source/blob/main/DECISIONLOG.md).
```
Comment (fivetran-jamie) [inline CHANGELOG.md:9]:
```suggestion
  - For more details, see the [DECISIONLOG entry](https://github.com/fivetran/dbt_tiktok_ads_source/blob/main/DECISIONLOG.md#age_groups-and-age-columns).
```

**Review by fivetran-jamie — COMMENTED** [review]
Looks pretty good -- left some questions and comments

File: CHANGELOG.md (hunk 3 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md]: Good call. Updated!

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-jamie Thanks for reviewing! I updated with your suggestions.

File: DECISIONLOG.md (hunk 4 above)
Comment (fivetran-catfritz) [reply DECISIONLOG.md:21]: oops removed.

**Pull request merged**